server/routerlicious/packages/protocol-base/src/protocol.ts

```diff
@@ -116,22 +149,49 @@ export class ProtocolOpHandler {
         this.minimumSequenceNumber = message.minimumSequenceNumber;
         this.sequenceNumber = message.sequenceNumber;
 
+        this.cachedProtocolState.minimumSequenceNumber = this.minimumSequenceNumber;
```

Comment:
What is the reason to have two versions of MSN / sequenceNumber?
When / how they are different? I see, for example, that updateMinimumSequenceNumber() updates only one value but not another one.
Have some comments on that above cachedProtocolState declaration would be usedful

Reply:
That said, I think updateMinimumSequenceNumber() should not touch this.minimumSequenceNumber - it's already updated above, and that's the only call to updateMinimumSequenceNumber(). I think updateMinimumSequenceNumber() should not have any arguments and work off current object state.